### PR TITLE
Add the column 'Params %'

### PR DESCRIPTION
**Pull request by luke396**
To fix [Idea: add the column &quot;Params %&quot; in addition to &quot;Params #&quot; · Issue #159 ](https://github.com/TylerYep/torchinfo/issues/159)

I am still a newbie and would like to get any advice and guidance!

### CODE REVIEW

File: torchinfo/enums.py [hunk 1]
```diff
@@ -29,6 +29,7 @@ class ColumnSettings(str, Enum):
     INPUT_SIZE = "input_size"
     OUTPUT_SIZE = "output_size"
     NUM_PARAMS = "num_params"
+    PER_PARAMS = "per_params"
```
Comment (TylerYep) [inline torchinfo/enums.py:32]: use the full name PARAMS_PERCENT = "params_percent"

File: torchinfo/formatting.py [hunk 2]
```diff
@@ -118,12 +122,13 @@ def layer_info_to_row(self, layer_info: LayerInfo, reached_max_depth: bool) -> s
                 new_line += self.format_row(f"{prefix}{inner_name}", inner_layer_info)
         return new_line
 
-    def layers_to_str(self, summary_list: list[LayerInfo]) -> str:
+    def layers_to_str(self, layers_summary: Any) -> str:
```
Comment (TylerYep) [inline torchinfo/formatting.py:125]: No need to change this. We shouldn't pass total_params as a parameter. Just use it in self.total_params

File: tests/test_output/basic_per_summary.out [hunk 3]
```diff
@@ -0,0 +1,14 @@
+==========================================================================================
+Layer (type:depth-idx)                   Param #                   Param %
+==========================================================================================
+SingleInputNet                           --                        0.00
```
Comment (TylerYep) [inline tests/test_output/basic_per_summary.out:4]: Let's add a percent sign to all of these?

Also, how should we handle decimal precision? Two decimals look okay here but might not work for large models

**Review by TylerYep — CHANGES_REQUESTED** [review]

File: tests/test_output/basic_per_summary.out (hunk 3 above)
Comment (luke396) [reply tests/test_output/basic_per_summary.out:4]: How about change two decimals to one decimal?

File: torchinfo/formatting.py (hunk 2 above)
Comment (luke396) [reply torchinfo/formatting.py:125]: Don't pass the total_params as parameter? We should add total_params to summary_list? Or some other ideas? I really don't think the current one is a good implementation.

File: torchinfo/layer_info.py [hunk 4]
```diff
@@ -291,6 +291,13 @@ def num_params_to_str(self, reached_max_depth: bool) -> str:
         leftover_params = self.leftover_params()
         return f"{leftover_params:,}" if leftover_params > 0 else "--"
 
+    def per_params_to_str(self, total_params: int) -> str:
+        leftover_params = self.leftover_params()
```
Comment (TylerYep) [inline torchinfo/layer_info.py:295]: This implementation doesn't work - leftover params does not include the number of parameters below this node.

File: torchinfo/formatting.py (hunk 2 above)
Comment (TylerYep) [reply torchinfo/formatting.py]: Oh I didn't realize total_params was not on this class. Ignore this comment

File: tests/test_output/basic_per_summary.out (hunk 3 above)
Comment (TylerYep) [reply tests/test_output/basic_per_summary.out]: Two decimals is fine for now, but in the future users might want to increase the precision. This seems like a good starting point though

File: torchinfo/layer_info.py (hunk 4 above)
Comment (TylerYep) [reply torchinfo/layer_info.py]: Fixed it to match the above function

**Review by TylerYep — APPROVED** [review]

**Comment by TylerYep** [comment]
Made some tweaks to have the calculations match and also right-justified the text so it looks nicer. Otherwise looks good, thanks for the contribution!

**Comment by luke396** [comment]
@TylerYep, Thanks your review a lot! I read your code implementation and learned a lot. My superficial attempts do have a lot of problems. If you don't mind, I will continue to try to solve issues and make PR in the future.

**Comment by TylerYep** [comment]
That would be great! All contributions are welcome :)

**Pull request merged**